esp-mbedtls/src/lib.rs

```diff
@@ -125,6 +141,22 @@ impl<T> Session<T> {
                 return Err(TlsError::OutOfMemory);
             }
 
+            let client_crt =
+                calloc(1, size_of::<mbedtls_x509_crt>() as u32) as *mut mbedtls_x509_crt;
+            if client_crt.is_null() {
+                free(ssl_context as *const _);
+                free(ssl_config as *const _);
+                return Err(TlsError::OutOfMemory);
+            }
+
+            let private_key =
+                calloc(1, size_of::<mbedtls_pk_context>() as u32) as *mut mbedtls_pk_context;
+            if private_key.is_null() {
+                free(ssl_context as *const _);
```

Comment:
if we end-up here we already successfully allocated memory for `client-crt` so we need to free that, too